### PR TITLE
zigs: remove `%give` and `%take` `to-account`

### PR DESCRIPTION
**Pull request by hosted-fornet**
Just putting some code up so we have something more concrete to discuss. If we decide we want to go forward with this I will make similar changes to fungible. This may also be better directed to next/smart?

```
>   "gas cost: 20.417"
```

### CODE REVIEW

**Comment by dr-frmr** [comment]
Been looking at this change and I think it's worth it. Also makes other contracts able to compose with tokens more easily: right now, contracts that want to execute %gives and %takes need to take in token account IDs and use them. This change simplifies not only the token standard, but every other contract that interacts with it.

**Comment by dr-frmr** [comment]
@hosted-fornet are you cool if I start committing to this branch with the various changes needed to %wallet and other contracts?

**Comment by hosted-fornet** [comment]
@dr-frmr Go for it: consider the PR yours.

**Comment by dr-frmr** [comment]
This is ready for review @hosted-fornet @tadad -- surprisingly few breakages/dependencies were disrupted, thought there'd be more stuff. The breaking changes affect the AMM contract and pokur escrow contract, which I will change

**Review by tadad — APPROVED** [review]
looks good - code is a lot cleaner now

File: con/lib/fungible.hoon [hunk 1]
```diff
@@ -126,53 +114,47 @@
     ^-  (quip call diff)
     =+  (need (scry-state from-account.act))
```
Comment (hosted-fornet) [inline con/lib/fungible.hoon:115]: comparing to zigs.hoon, there is a line missing (zigs.hoon reads):

```hoon
    ?>  !=(to.act id.caller.context)
    =+  (need (scry-state from-account.act))
```

Should we be doing that assert here as well?

File: con/lib/fungible.hoon [hunk 2]
```diff
@@ -126,53 +114,47 @@
...
     =/  giver  (husk account:sur - `this.context `id.caller.context)
```
Comment (hosted-fornet) [inline con/lib/fungible.hoon:116]: nit: for consistency with zigs.hoon, would recommend changing this line to

```hoon
    =+  giver=(husk account:sur - `this.context `id.caller.context)
```

File: con/lib/fungible.hoon [hunk 3]
```diff
@@ -229,21 +210,19 @@
     ?>  .=(nonce.act -)
```
Comment (hosted-fornet) [inline con/lib/fungible.hoon:210]: nit: use form

```hoon
?> .=(nonce.act -)
```

**Review by hosted-fornet — COMMENTED** [review]
Mostly nits, otherwise LGTM. Didn't test.

File: con/lib/fungible.hoon (hunk 1 above)
Comment (dr-frmr) [reply con/lib/fungible.hoon:115]: Ah, yes.

**Comment by dr-frmr** [comment]
Merging. This PR also provides a bug fix to these token contracts that allowed people to send tokens to themselves and double their balances. Important to fix!

**Pull request merged**